hybridq/architecture/google/sycamore.py

```diff
@@ -0,0 +1,165 @@
+"""
+Author: Salvatore Mandra (salvatore.mandra@nasa.gov)
+
+Copyright © 2021, United States Government, as represented by the Administrator
+of the National Aeronautics and Space Administration. All rights reserved.
+
+The HybridQ: A Hybrid Simulator for Quantum Circuits platform is licensed under
+the Apache License, Version 2.0 (the "License"); you may not use this file
+except in compliance with the License. You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0.
+
+Unless required by applicable law or agreed to in writing, software distributed
+under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+
+Types
+-----
+**`Qubit`**: `tuple[int, int]`
+
+**`QpuLayout`**: `list[Qubit]`
+
+**`Coupling`**: `tuple[Qubit, Qubit]`
+
+Attributes
+----------
+drawing: str
+    Drawing of the Google Sycamore QPU.
+
+layout: QpuLayout
+    Qubits available in Google Sycamore QPU.
+
+couplings: list[Coupling]
+    All couplings available in Google Sycamore QPU.
+
+get_layers: callable[QpuLayout] -> dict[str, list[Coupling]]
+    Given a `QpuLayout` returns layers of couplings as defined in
+    "Quantum supremacy using a programmable superconducting processor"
+    [Nature 574 (7779), 505-510].
+"""
+
+from __future__ import annotations
+from typing import List, Tuple, Callable
+from hybridq.architecture.utils import get_layout
+from hybridq.utils import sort, argsort
+
+__all__ = ['drawing', 'layout', 'couplings', 'get_layers']
+
+# Define Qubit type
+Qubit = Tuple[int, int]
+
+# Define Coupling
+Coupling = Tuple[Qubit, Qubit]
+
+# Define QpuLayout
+QpuLayout = List[Qubit]
+
+# Drawing
+drawing = r"""
+      X-X
+      | |
+    X-X-X-X
+    | | | |
+  X-X-X-X-X-X
+  | | | | | |
+X-X-X-X-X-X-X-X
+| | | | | | | |
+X-X-X-X-X-X-X-X-X
+  | | | | | | | |
+  X-X-X-X-X-X-X-X-X
+      | | | | | |
+      X-X-X-X-X-X
+      | | | | |
+      X-X-X-X-X
+        | | |
+        X-X-X
+          |
+          X
+"""
+
+# Get couplings and layout
+layout, couplings = get_layout(drawing)
+
+# Check for consistency
+from hybridq.extras.architecture.sycamore.sycamore import gmon54 as __gmon54, \
+        get_all_couplings as __get_all_couplings
+assert (layout == __gmon54)
+assert (couplings == __get_all_couplings(__gmon54))
+
+
+def _in_simplifiable_layout(layout_idx: int) -> Callable[[Coupling], bool]:
```

Comment:
Why do you return a callable instead of a bool here? I'm sure there's a good reason, I just typically see in*/is* functions return a bool

Reply:
This is a very old designed .. fixed.